### ISSUE TITLE
padding-line-between-statements should support TSTypeAliasDeclaration

### ISSUE THREAD

**Issue by pascalgn** [eslint/eslint#13477]
**What rule do you want to change?**
padding-line-between-statements

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
Currently, there is no way to specify TS (TypeScript) elements in the padding-line-between-statements rule, specifically "interface" and "type" definitions.

**Please provide some example code that this change will affect:**

```ts
type SomeType = {
  value: string;
};
interface SomeInterface {
}
```

**What does the rule currently do for this code?**

Giving "type" or "interface" will throw an error about unsupported options.

**What will the rule do after it's changed?**

With the configuration

```json
      "rules": {
        "padding-line-between-statements": [
          "error",
          {
            "blankLine": "always",
            "prev": "type",
            "next": "interface"
          }
        ]
      }
```

the code example from the previous section should give an error. The following code section should not give that error anymore:

```ts
type SomeType = {
  value: string;
};

interface SomeInterface {
}
```

**Are you willing to submit a pull request to implement this change?**
Possibly

**Related issues**
- https://github.com/eslint/eslint/issues/9642 suggests a more generic solution, but did not get approval. This issue suggests a more specific rule change

**Comment by yeonjuan** [comment]
Hi @pascalgn :)
> Currently, there is no way to specify TS (TypeScript) elements in the padding-line-between-statements rule, specifically "interface" and "type" definitions.

It's because the ESLint core didn't support the typescript keyword. I suggest opening an issue in the [typescript-eslint repo](https://github.com/typescript-eslint/typescript-eslint).

**Comment by pascalgn** [comment]
Thanks for the pointer! Reading the FAQ there, I found the way forward for this issue: https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

So as far as I understand, the way to solve this issue would be to create a new extension rule, `@typescript-eslint/padding-line-between-statements` (it [doesn't exist yet](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/README.md#extension-rules)).

I've created https://github.com/typescript-eslint/typescript-eslint/issues/2293 to track the issue.

Thanks for your help!
